src/lsm/merge_iterator.zig

```diff
@@ -0,0 +1,103 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const util = @import("../util.zig");
+
+const k_max = 2;
+
+pub fn MergeIteratorType(
+    comptime Table: type,
+    comptime IteratorA: type,
+    comptime IteratorB: type,
+) type {
+    return struct {
+        const Self = @This();
+
+        iterator_a: *IteratorA,
+        iterator_b: *IteratorB,
+
+        empty_a: bool,
+        empty_b: bool,
+
+        previous_key_popped: ?Table.Key,
+
+        pub fn init(
+            iterator_a: *IteratorA,
+            iterator_b: *IteratorB,
+        ) Self {
+            return Self{
+                .iterator_a = iterator_a,
+                .iterator_b = iterator_b,
+                .empty_a = false,
+                .empty_b = false,
+                .previous_key_popped = null,
+            };
+        }
+
+        pub fn empty(it: Self) bool {
+            return it.empty_a and it.empty_b;
+        }
+
+        pub fn pop(it: *Self) ?Table.Value {
```

Comment:
:+1: 

I actually don't like this design (too much branching, plus errors are more expensive than they look), but it matches k_way_merge and I'll refrain from messing with the iterators too much until merging becomes a bottleneck.

Reply:
I can imagine. Once it's blocking perf we'll probably want to make it either push-based, or keep it pull-based but return `*const [C]Value` instead of `Value` to amortize the ovehead of the iterator control flow.